dive_authors.py

```diff
@@ -191,37 +145,85 @@ def get_authordateinfo_list(self):
 
         return self.authors_dateinfo
 
-    def get_latest_author_by_email(self, name):
-        if not hasattr(name, 'decode'):
-            name = str.encode(name)
 
-        name = name.decode("unicode_escape", "ignore")
-        return self.authors_by_email[name]
+def get_all_repos(access_token):
+    repos = []
+    page = 1
+    url = 'https://api.github.com/orgs/industrydive/repos?per_page=100&page=%s&access_token=%s'
+    while True:
+        response = requests.get(url % (page, access_token))
+
+        try:
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as e:
+            print 'Error: %s' % e
+
+        response_json = response.json()
+        for repo in response_json:
+            repos.append({
+                'name': repo['name'],
+                'ssh_url': repo['ssh_url'],
+                'pushed_at': repo['pushed_at'],
+            })
+        # break if we are on a page with < 100 repos (since we've reached the end)
+        if len(response_json) == 100:
+            page = page + 1
+        else:
+            break
+
+    return repos
+
+
+@click.command()
+@click.option(
+    '--year',
+    default=datetime.datetime.today().year - 1,
+    help='Year to run git-authors for. Defaults to the previous year',
+)
+@click.option(
+    '--access-token',
+    help='Create a Github access token',
+)
+@click.option(
+    '--outfile',
+    'outfile_name',
+    default='git_stats.csv',
+    help='File name to write the git stats to',
+    required=True,
+)
+def main(year, access_token, outfile_name):
+    script_path = os.path.dirname(os.path.abspath(__file__))
+    path_to_outfile = os.path.join(script_path, outfile_name)
 
-    def get_latest_email_by_author(self, name):
-        return self.emails_by_author[name]
+    # create a temporary directory
+    temp_dir_path = mkdtemp()
+    os.chdir(temp_dir_path)
 
-def is_valid_extension(string):
-    """
-    Hacky way to say all extensions are valid.
-    :param str string: The commit line.
-    :return: True
-    """
-    return True
+    try:
+        with open(path_to_outfile, 'wb') as outfile:
+            outwriter = csv.writer(outfile)
+            outwriter.writerow(COLUMN_HEADERS)
 
-def main():
-    for project_name in PROJECT_NAMES:
-        os.system('git clone git@github.com:industrydive/%s' % project_name)
-        repo_path = os.path.join(projects_path, project_name)
+            repos = get_all_repos(access_token)
 
-        runner = DiveRunner()
-        runner.repo = repo_path
-        runner.project_name = project_name
+            for repo in repos:
+                # if the repo has been pushed to after the first day of the target year
+                pushed_at_time = datetime.datetime.strptime(repo['pushed_at'], '%Y-%m-%dT%H:%M:%SZ')
+                if pushed_at_time > datetime.datetime(year, 1, 1):
```

Comment:
should this be `>=` instead of `>`? i think the way it's written it would only get things pushed on or after Jan. 2, right?

Reply:
`datetime.datetime(2019, 1, 1)` creates an object that is 2019-01-01 00:00:00, so it should get everything published before midnight. I guess I should add the `=` anyway though